### PR TITLE
2.0.0

### LINKED ISSUE #3: Use gremlins
A good candidate to use [gremlins](https://github.com/marmelab/gremlins.js) to stress test & capture bugs :sandal:

### PR DESCRIPTION
**Pull request by tristen**
- Pass `mode` & `colorAttribute` as optional properties to the component.
- Add `mode` & `colorAttribute` values to `onChange`.
- Remove localstorage management of color mode and attributes.
- Prefix general classnames with `cp` 
- Simplify the demo page by pointing to docs in the reame
- Tests. Close #3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -26,57 +26,63 @@ module.exports = React.createClass({
...
-      mode: ls.get('mode') ? ls.get('mode') : 'rgb',
-      colorMode: ls.get('colorMode') ? ls.get('colorMode') : 'h'
+      mode: this.props.mode ? this.props.mode : 'rgb',
+      colorAttribute: this.props.colorAttribute ? this.props.colorAttribute : 'h'
     };
   },
 
+  emitOnChange: function(change) {
+    var { color, mode, colorAttribute } = this.state;
+    this.props.onChange(Object.assign(
```
Comment (tmcw) [inline index.js]: this library already includes xtend. why this too?

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "stage": 0,
+  "loose": "all",
+  "plugins": ["object-assign"]
```
Comment (tmcw) [inline .babelrc]: Do not polyfill globals in libraries.

File: index.js [hunk 3]
```diff
@@ -335,33 +341,33 @@ module.exports = React.createClass({
...
-                    onFocus={this.colorMode.bind(null, 's')}
+                    onFocus={this.setColorAttribute.bind(null, 's')}
                     onChange={this.changeHSV.bind(null, 's')}
                     type='number'
                     min={0}
                     max={100}
                     step={1} />
                 </fieldset>
-                <fieldset className={(colorMode === 'v') ? 'active' : ''}>
+                <fieldset className={(colorAttribute === 'v') ? 'active' : ''}>
```
Comment (tmcw) [inline index.js]: unnecessary parens

File: example/index.js [hunk 4]
```diff
@@ -28,6 +30,8 @@ var App = React.createClass({
   },
 
   onChange: function(color) {
+    console.log('onChange: ', color);
```
Comment (tmcw) [inline example/index.js]: remove this

File: index.js [hunk 5]
```diff
@@ -162,32 +162,34 @@ module.exports = React.createClass({
...
+    this.setState(obj);
+    this.emitOnChange(obj);
   },
 
-  setMode: function(e) {
-    ls.set('mode', e.target.value);
-    this.setState({mode: e.target.value});
+  setColorAttribute: function(attribute) {
+    var obj = { colorAttribute: attribute };
+    this.setState(obj);
```
Comment (tmcw) [inline index.js:170]: setState is async and is treated here and in many other places as sync.

**Comment by tmcw** [comment]
I don't understand what's the point of switching from xtend to Object.assign and from browserify to webpack. The problem that we have with react-colorpickr is, and has always been, robustness, and this PR does not contribute to that goal at all: this is still an untested component.

File: example/index.js (hunk 4 above)
Comment (tristen) [inline example/index.js]: It's on the example page and helpful to see what the onChange output looks like when evaluating this component no?

**Pull request merged**